pkg/operator/options/options_validation.go

```diff
@@ -0,0 +1,65 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/go-playground/validator/v10"
+	"go.uber.org/multierr"
+)
+
+func (o Options) Validate() error {
+	validate := validator.New()
+	return multierr.Combine(
+		//o.validateRequiredFields(),	// TODO: add back to Validate when karpenter-global-settings (and merge logic) are completely removed
+		o.validateEndpoint(),
+		o.validateVMMemoryOverheadPercent(),
+		validate.Struct(o),
+	)
+}
+
+func (o Options) validateEndpoint() error {
+	if o.ClusterEndpoint == "" {
+		return nil
+	}
+	endpoint, err := url.Parse(o.ClusterEndpoint)
+	// url.Parse() will accept a lot of input without error; make
+	// sure it's a real URL
+	if err != nil || !endpoint.IsAbs() || endpoint.Hostname() == "" {
+		return fmt.Errorf("\"%s\" not a valid clusterEndpoint URL", o.ClusterEndpoint)
+	}
+	return nil
+}
+
+func (o Options) validateVMMemoryOverheadPercent() error {
+	if o.VMMemoryOverheadPercent < 0 {
+		return fmt.Errorf("vm-memory-overhead-percent cannot be negative")
+	}
+	return nil
+}
+
+func (o Options) validateRequiredFields() error {
+	if o.ClusterEndpoint == "" {
+		return fmt.Errorf("missing field, cluster-endpoint")
+	}
+	if o.ClusterName == "" {
+		return fmt.Errorf("missing field, cluster-name")
+	}
+	return nil
```

Comment:
`kubelet-bootstrap-token` and `ssh-public-key` are also required?

Reply:
Will add it. Looks like we were missing this up until this point...